hibernate-core/src/main/java/org/hibernate/engine/config/spi/ConfigurationService.java

```diff
@@ -78,6 +81,6 @@ interface Converter<T> {
 		 *
 		 * @return The converted (typed) value.
 		 */
-		T convert(Object value);
+		@PolyNull T convert(@PolyNull Object value);
```

Comment:
We said we'll make the converter non-null, so please revert this.

```suggestion
		T convert(Object value);
```

Reply:
I only pushed this to show what works in the current situation, I'll revert it as soon as we figure out how to make it actually work without the @PolyNulls on the converter